### PR TITLE
raise decode error instead of ContentDecodingError

### PR DESCRIPTION
**Pull request by xiangyan99**

### CODE REVIEW

File: sdk/core/azure-core/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## 1.15.1 (Unreleased)
 
+### Breaking Changes
+
+- Sync stream downloading now raises `azure.core.exceptions.DecodeError` rather than `requests.exceptions.ContentDecodingError`
```
Comment (annatisch) [inline sdk/core/azure-core/CHANGELOG.md:7]: Are we okay with this breaking change because no one should have written code to explicitly handle this error?

Comment (xiangyan99) [reply sdk/core/azure-core/CHANGELOG.md:7]: Yes. Checked with @johanste and he agreed this was an acceptable breaking change.

File: sdk/core/azure-core/azure/core/pipeline/transport/_requests_basic.py [hunk 2]
```diff
@@ -58,11 +59,11 @@ def _read_raw_stream(response, chunk_size=1):
             for chunk in response.raw.stream(chunk_size, decode_content=False):
                 yield chunk
         except ProtocolError as e:
-            raise requests.exceptions.ChunkedEncodingError(e)
-        except DecodeError as e:
-            raise requests.exceptions.ContentDecodingError(e)
+            raise ServiceResponseError(e, error=e)
+        except CoreDecodeError as e:
+            raise DecodeError(e, error=e)
```
Comment (annatisch) [inline sdk/core/azure-core/azure/core/pipeline/transport/_requests_basic.py:64]: This scenario is being correctly handled in the other transports?

Comment (xiangyan99) [reply sdk/core/azure-core/azure/core/pipeline/transport/_requests_basic.py:64]: Currently different transport raises different type of error. (That's the reason we need this change).

We want to uniform the errors so user can catch all by using except AzureError.

**Review by lmazuel — APPROVED** [review]

**Pull request merged**